client/src/api/v1/index.ts

```diff
@@ -82,11 +82,19 @@ export default class ServerApi {
     }
   }
 
-  protected async put<ResponseDataType>(
-    path: string
+  protected async put<ResponseDataType, RequestDataType=undefined>(
+    path: string, form?: RequestDataType
   ): Promise<{ status: number; data?: ResponseDataType }> {
     try {
-      const response = await this.api.put(path);
+      let response;
```

Comment:
Could we keep it consistent with how we did the previous functions and just set to empty object if there is no form?